app/lib/actions/pulp/repository/sync.rb

```diff
@@ -66,7 +66,12 @@ def contents_changed?(tasks)
           end
 
           if sync_task && sync_task['state'] == 'finished' && sync_task[:result]
-            sync_task['result']['added_count'] > 0 || sync_task['result']['removed_count'] > 0 || sync_task['result']['updated_count'] > 0
+            if sync_task['result']['added_count'] > 0 || sync_task['result']['removed_count'] > 0 || sync_task['result']['updated_count'] > 0
+              true
+            else
+              repo = ::Katello::Repository.find(sync_task['result']['repo_id'])
+              repo ? repo.pulp_counts_differ? : true
```

Comment:
Yeah, i think its better to assume something has changed than not :+1: 

Will rereview soon